src/test/kotlin/uk/gov/justice/digital/hmpps/approvedpremisesapi/integration/cas1/seed/SeedFromXLSXApprovedPremisesRoomsTest.kt

```diff
@@ -104,47 +112,115 @@ class SeedFromXLSXApprovedPremisesRoomsTest : SeedTestBase() {
       "example.xlsx",
     )
 
-    val room1 = roomRepository.findByCode("SWABI01NEW")
+    val room1 = roomRepository.findByCode("Q999 - 1")
     assertThat(room1!!.characteristics).isEmpty()
 
-    val bed1 = bedRepository.findByCodeAndRoomId("SWABI01NEW - 1", room1.id)
-    assertThat(bed1!!.name).isEqualTo("SWABI01NEW")
+    val bed1 = bedRepository.findByCodeAndRoomId("SWABI01NEW", room1.id)
+    assertThat(bed1!!.name).isEqualTo("1")
     assertThat(
       bed1.room.id == room1.id &&
-        bed1.room.code == "SWABI01NEW",
+        bed1.room.code == "Q999 - 1",
     )
 
-    val room2 = roomRepository.findByCode("SWABI02NEW")
+    val room2 = roomRepository.findByCode("Q999 - 2")
     assertThat(room2!!.characteristics).anyMatch {
       it.name == "Is this room located on the ground floor?" &&
         it.propertyName == "isGroundFloor"
     }
 
-    val bed2 = bedRepository.findByCodeAndRoomId("SWABI02NEW - 1", room2.id)
-    assertThat(bed2!!.name).isEqualTo("SWABI02NEW")
+    val bed2 = bedRepository.findByCodeAndRoomId("SWABI02NEW", room2.id)
+    assertThat(bed2!!.name).isEqualTo("1")
     assertThat(
       bed2.room.id == room2.id &&
-        bed2.room.code == "SWABI02NEW",
+        bed2.room.code == "Q999 - 2",
     )
 
-    val room3 = roomRepository.findByCode("SWABI03NEW")
+    val room3 = roomRepository.findByCode("Q999 - 3")
     assertThat(room3!!.characteristics).isEmpty()
 
-    val bed3 = bedRepository.findByCodeAndRoomId("SWABI03NEW - 1", room3.id)
-    assertThat(bed3!!.name).isEqualTo("SWABI03NEW")
+    val bed3 = bedRepository.findByCodeAndRoomId("SWABI03NEW", room3.id)
+    assertThat(bed3!!.name).isEqualTo("1")
     assertThat(
       bed3.room.id == room3.id &&
-        bed3.room.code == "SWABI03NEW",
+        bed3.room.code == "Q999 - 1",
+    )
+  }
+
+  @Test
+  fun `Creating two new rooms and three new beds with a characteristic succeeds`() {
+    val localAuthorityArea = localAuthorityEntityFactory.produceAndPersist()
+    val probationRegion = probationRegionEntityFactory.produceAndPersist()
+    val qCode = "Q999"
+    val premises = approvedPremisesEntityFactory.produceAndPersist {
+      withLocalAuthorityArea(localAuthorityArea)
+      withProbationRegion(probationRegion)
+      withQCode(qCode)
+    }
+    val premisesId = premises.id
+
+    val header = listOf("Unique Reference Number for Bed", "SWABI01NEW", "SWABI02NEW", "SWABI03NEW")
+    val rows = mutableListOf(
+      "Room Number / Name",
+      "1",
+      "2",
+      "2",
+      "Bed Number (in this room i.e if this is a single room insert 1.  If this is a shared room separate entries will need to be made for bed 1 and bed 2)",
+      "1",
+      "1",
+      "2",
+    )
+    rows.addCharacteristics(3, mapOf("Is this room located on the ground floor?" to listOf(1, 2)))
```

Comment:
it would be really helpful/more readable if we added arg names when using addCharacteristics

Reply:
Would this be more readable like:

```
rows.addCharacteristics("Is this room located on the ground floor?", listOf("N","Y","Y"))
```

Reply:
Will revisit this when work is done to address [APS-1757](https://dsdmoj.atlassian.net/browse/APS-1757).